lib/licensee/license.rb

```diff
@@ -144,20 +138,22 @@ def name_without_version
     def title_regex
       return @title_regex if defined? @title_regex
 
-      title_regex = ALT_TITLE_REGEX[key]
-      title_regex ||= begin
-        string = name.downcase.sub('*', 'u')
-        string.sub!(/\Athe /i, '')
-        string.sub!(/,? version /, ' ')
-        string.sub!(/v(\d+\.\d+)/, '\1')
-        string = Regexp.escape(string)
-        string = string.sub(/\\ licen[sc]e/i, '(?:\ licen[sc]e)?')
-        string = string.sub(/\\ (\d+\\.\d+)/, ',?\s+(?:version\ |v(?:\. )?)?\1')
-        string = string.sub(/\bgnu\\ /, '(?:GNU )?')
-        Regexp.new string, 'i'
-      end
-
-      parts = [title_regex, key]
+      string = name.downcase.sub('*', 'u')
+      string.sub!(/\Athe /i, '')
+      string.sub!(/,? version /, ' ')
+      string.sub!(/v(\d+\.\d+)/, '\1')
+      string = Regexp.escape(string)
+      string = string.sub(/\\ licen[sc]e/i, '(?:\ licen[sc]e)?')
+      string = string.sub(/\\ (\d+\\.\d+)/, ',?\s+(?:version\ |v(?:\. )?)?\1')
+      string = string.sub(/\bgnu\\ /, '(?:GNU )?')
+      title_regex = Regexp.new string, 'i'
+
+      string = key.sub('-', '[- ]')
+      string.sub!('.', '\.')
```

Comment:
Using `Regexp.escape` as the first step should be a bit more robust and consistent with the title treatment above.